src/frontend/src/flows/authenticate/index.ts

```diff
@@ -1,229 +1,38 @@
-        register,
-        recoverAnchor: () => recoverAnchor(anchorInput.readUserNumber()),
-      })}
-    </div>
-    ${footer}`;
-};
-
-export const authenticatePage = (
-  connection: Connection,
-  authContext: AuthContext
-): Promise<LoginData> => {
-  const retryOnError = async (result: LoginFlowResult): Promise<LoginData> => {
-    switch (result.tag) {
-      case "err":
-        await displayError({
-          title: result.title,
-          message: result.message,
-          detail: result.detail !== "" ? result.detail : undefined,
-          primaryButton: "Try again",
-        });
-        return authenticatePage(connection, authContext);
-      case "ok":
-        return {
-          userNumber: result.userNumber,
-          connection: result.connection,
-        };
-      case "canceled":
-        return authenticatePage(connection, authContext);
-      default:
-        unreachable(result);
-        break;
-    }
-  };
-
-  return new Promise<LoginData>((resolve) => {
-    displayPage({
-      origin: authContext.requestOrigin,
-      onContinue: async (userNumber) => {
-        const loginData = await authenticate(connection, userNumber).then(
-          retryOnError
-        );
-        setUserNumber(loginData.userNumber);
-        resolve(loginData);
-      },
-      register: async () => {
-        const loginData = await registerIfAllowed(connection).then(
-          retryOnError
-        );
-        setUserNumber(loginData.userNumber);
-        resolve(loginData);
-      },
-      recoverAnchor: (userNumber) => useRecovery(connection, userNumber),
-      userNumber: getUserNumber(),
-      derivationOrigin: authContext.authRequest.derivationOrigin,
-    });
-  });
-};
-
-export const authenticate = async (
-  connection: Connection,
-  userNumber: bigint
-): Promise<LoginFlowSuccess | LoginFlowError> => {
-  try {
-    const result = await withLoader(() => connection.login(userNumber));
-    return apiResultToLoginFlowResult(result);
-  } catch (error) {
-    return {
-      tag: "err",
-      title: "Authentication Failed",
-      message: "An error occurred during authentication.",
-      detail: error instanceof Error ? error.message : JSON.stringify(error),
-    };
-  }
-};
-
-export const displayPage = (props: PageProps): void => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(props), container);
+}: {
+  origin: string;
+  derivationOrigin?: string;
+}): AuthTemplates => {
+  const message = html`<p class="t-lead">
+    Connect to<br />
+    <span class="t-strong">${origin}</span><br />
+  </p>`;
+  const chasm =
+    derivationOrigin !== undefined
+      ? {
+          info: "shared identity",
+          message: html`<span class="t-strong">${origin}</span> is an
+            alternative domain of <br /><span class="t-strong"
+              >${derivationOrigin}</span
+            ><br />and you will be authenticated to both with the same identity. `,
```

Comment:
the space after `identity. ` is needed? (most important question of the day I know...)